main.ts

```diff
@@ -17,127 +14,148 @@ const simpleSliderTemplate = (source: string) => `
 `;
 
 const createDOMElFromString = (domstring: string): HTMLElement => {
- const html = new DOMParser().parseFromString(domstring, "text/html");
- return <HTMLElement>html.body.firstChild;
+    const html = new DOMParser().parseFromString(domstring, "text/html");
+    return <HTMLElement>html.body.firstChild;
 };
 
 // Slider Classes
 abstract class BaseSlider {
- protected imagePath = "./assets/";
-
- public currentIndex = 0;
- public el: HTMLElement;
- protected slides: HTMLElement[] = [];
-
- // TODO: add union type for slides
- abstract addSlides(slides: string[]): void;
-
- constructor(selector: string, private showButtons: boolean = true) {
-   this.el = document.querySelector(selector);
- }
-
- // TODO: next and previous methods need to be fixed
- // when slides index > slides.length you can see empty slide
-
- public next(): void {
-   this.showSlide(++this.currentIndex);
- }
-
- public previous(): void {
-   this.showSlide(--this.currentIndex);
- }
-
- protected showSlide(index: number): void {
-   this.slides.forEach((item, i) => {
-     if (index === i) {
-       item.classList.remove("is-hidden");
-       return;
-     }
-
-     item.classList.add("is-hidden");
-   });
- }
-
- private addButtons(): void {
-   const buttonsContainer = document.createElement("div");
-   buttonsContainer.classList.add("navigation");
-   buttonsContainer.classList.add("has-text-centered");
-   buttonsContainer.innerHTML = `
+    protected imagePath = "./assets/";
+
+    public currentIndex: number = 0;
+    public el: HTMLElement;
+    protected slides: HTMLElement[] = [];
+
+    // TODO: add union type for slides
+    abstract addSlides(slides: string[] | ITextSlide[]): void;
+
+    constructor(selector: string, private showButtons: boolean = true) {
+        this.el = document.querySelector(selector);
+    }
+
+    // TODO: next and previous methods need to be fixed
+    // when slides index > slides.length you can see empty slide
+
+    public next(): void {
+        if(this.currentIndex === this.slides.length - 1){
+            this.currentIndex = -1;
```

Comment:
A little bit confusing) 
You can just set the index and pass it to the `showSlide()`

Anyways works as expected :-) 
One another approach of looping